src/femr/labelers/omop_inpatient_admissions.py

```diff
@@ -143,11 +148,28 @@ def __init__(
         self.prediction_time_adjustment_func = prediction_time_adjustment_func
 
     def label(self, patient: Patient) -> List[Label]:
-        """Label all admissions with admission length > `self.long_time`"""
+        """
+        Label all admissions with admission length > `self.long_time`
+        Exclude admission if time of discharge or death < prediction time
+        """
         labels: List[Label] = []
         for admission_time, discharge_time in get_inpatient_admission_discharge_times(patient, self.ontology):
             is_long_admission: bool = (discharge_time - admission_time) >= self.long_time
             prediction_time: datetime.datetime = self.prediction_time_adjustment_func(admission_time)
+
+            # exclude if discharge or death occurred before prediction time
+            death_concepts: Set[str] = map_omop_concept_codes_to_femr_codes(self.ontology, get_death_concepts())
+            death_times: List[datetime.datetime] = []
+            for e in patient.events:
+                if e.code in death_concepts:
+                    death_times.append(e.start)
+
+            if prediction_time > discharge_time:
```

Comment:
Isn't this duplicative with the modifications to get prediction times?

Reply:
I don't think so. The prediction time is the adjusted admission time, and this checks whether the adjustment has moved the prediction time to > discharge time, and if so, exclude.

Reply:
Actually, I take that back... what I just changed is incorrect. Will think it through and consider your suggestion.